iis-wf/iis-wf-export-actionmanager/src/test/java/eu/dnetlib/iis/wf/export/actionmanager/sequencefile/FieldAccessor.java

```diff
@@ -0,0 +1,143 @@
+						}
+						Object fieldValue = getFieldValue(fieldName, currentObject);
+						if (decoder.canHandle(fieldValue)) {
+							currentObject = decoder.decode(fieldValue);
+						} else {
+							throw new Exception(fieldValue!=null?
+									decoder.getClass().getName() + " decoder is not capable of handling object: " + 
+									fieldValue.toString() + ", which is " + fieldValue.getClass().getName() + " class instance"
+									:decoder.getClass().getName() + " decoder is not capable of handling null object");
+						}
+					} else if (token.indexOf('[')>0) {
+						String fieldName = token.substring(0, token.indexOf('['));
+						currentObject = getFieldValue(fieldName, currentObject);
+						if (currentObject!=null) {
+							 if (currentObject.getClass().isArray()) {
+								 currentObject = Array.get(currentObject, getArrayPositionFromToken(token));	 
+							 } else if (List.class.isAssignableFrom(currentObject.getClass())) {
+								 currentObject = ((List<?>)currentObject).get(getArrayPositionFromToken(token));
+							 } else {
+								 throw new Exception("unable to extract array or list element "
+											+ "based on field path: " + fieldPath + ", "
+											+ "object is neither an array nor list:" + 
+											currentObject.getClass().getName());
+							 }
+						} else {
+							throw new Exception("unable to extract array or list element "
+									+ "based on field path: " + fieldPath + ", "
+									+ "object is null!");
+						}
+					} else {
+						currentObject = getFieldValue(token, currentObject);
+					}
+				}
+				return currentObject;
+			} else {
+				return null;
+			}
+		} catch (Exception e) {
+//			we are extremely verbose to make log analysis process easier 
+			throw new Exception("unable to resolve field path '" + fieldPath 
+					+ "' on object " + object, e);
+		}
+	}
+
+	/**
+	 * Extracts array/list position from given token
+	 * @param token
+	 * @return extracted position
+	 */
+	private static int getArrayPositionFromToken(String token) {
+		return Integer.parseInt(StringUtils.substringBetween(token, "[", "]"));
+	}
+	
+	/**
+	 * Retrieves value from given object by executing getter method on given field using reflection.
+	 * @param fieldName name of the field holding value to be returned
+	 * @param object object holding given value inside
+	 * @return field value obtained from given object
+	 * @throws Exception
+	 */
+	private static Object getFieldValue(String fieldName, Object object) throws Exception {
+		Method method = getMethod(object.getClass(), GETTER + fieldName);
+		return method.invoke(object);
+	}
+
+	/**
+	 * Provides {@link Method} for given class and method name
+	 * @param clazz class from which method should be returned
+	 * @param name method name
+	 * @return declated method
+	 * @throws NoSuchFieldException
+	 * @throws NoSuchMethodException
+	 */
+	private static Method getMethod(Class<?> clazz, String name) throws NoSuchFieldException, NoSuchMethodException {
```

Comment:
`clazz.getMethod()` would work for public getters. Do we need to find non-public getters?